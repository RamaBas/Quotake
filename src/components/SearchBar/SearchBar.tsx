import React from 'react';
import { TextField, Box } from '@mui/material';
import { useTranslation } from 'react-i18next';

interface SearchBarProps {
  searchTerm: string;
  onSearchChange: (term: string) => void;
}

const SearchBar: React.FC<SearchBarProps> = ({ 
  searchTerm, 
  onSearchChange,
}) => {
  const { t } = useTranslation();

  return (
    <Box 
      sx={{ 
        mb: 3, 
        display: 'flex', 
        gap: 2,
        alignItems: 'center'
      }}
    >
      <TextField
        label={t('searchPlaceholder')}
        value={searchTerm}
        onChange={(e) => onSearchChange(e.target.value)}
        fullWidth
        variant="outlined"
        size="small"
        sx={{ 
          backgroundColor: '#fff', 
          borderRadius: 1, 
          boxShadow: 1,
          flexGrow: 1
        }}
      />
    </Box>
  );
};

export default SearchBar;