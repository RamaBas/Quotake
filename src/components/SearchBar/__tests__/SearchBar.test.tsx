import { render, screen, fireEvent } from '@testing-library/react';
import SearchBar from '../SearchBar';


describe('SearchBar Component', () => {
  const mockSearchChange = jest.fn();
  const initialProps = {
    searchTerm: '',
    onSearchChange: mockSearchChange,
  };

  it('renders correctly with initial props', () => {
    const { asFragment } = render(<SearchBar {...initialProps} />);

    
    const textField = screen.getByRole('textbox');
    expect(textField).toBeInTheDocument();
    expect(textField).toHaveValue('');

    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onSearchChange when typing', () => {
    render(<SearchBar {...initialProps} />);
    
    const textField = screen.getByRole('textbox');
    const testValue = 'test search';
    
    fireEvent.change(textField, { target: { value: testValue } });
    
    expect(mockSearchChange).toHaveBeenCalledTimes(1);
    expect(mockSearchChange).toHaveBeenCalledWith(testValue);
  });
});