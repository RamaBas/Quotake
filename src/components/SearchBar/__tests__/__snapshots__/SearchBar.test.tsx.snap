// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component renders correctly with initial props 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-18pzm0c"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-u0wg97-MuiFormControl-root-MuiTextField-root"
      data-testid="search-bar-input"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1kfabtt-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="«r0»"
        id="«r0»-label"
      >
        searchPlaceholder
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1pzfmz2-MuiInputBase-input-MuiOutlinedInput-input"
          id="«r0»"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-81qg8w"
          >
            <span>
              searchPlaceholder
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
